### PR TITLE
Update golang.md

### PR DESCRIPTION
**Pull request by letzya**
Based on a client's feedback - improve the titles and the wording so users can find our instructions for 5.1 onwards
 
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

---

### Preview Link

<br>

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [ ] I have added a **preview link** to the PR description.
- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I **labelled** the PR

### CODE REVIEW

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 1]
```diff
@@ -33,35 +33,35 @@ It's also possible to access the API definition data structure from within a plu
...
+
+
 ### Initialise plugin for Gateway v5.1 and above (v5.1+)
 
-In Gateway version 5.1, the gateway and plugins transitioned to a go
-module build and don't use vendoring anymore. To create a full workspace
-follow the guide below. To find out more about Go workspaces, visit the [official documentation](https://go.dev/ref/mod#workspaces).
+**If you are using Gateway version 5.1 or higher, please follow the steps outlined in this section. These instructions are tailored to the latest Gateway software.**
+
+In Gateway version 5.1, the gateway and plugins transitioned to a [Go modules](https://go.dev/ref/mod#introduction) build and don't use [govendor](https://pkg.go.dev/github.com/kardianos/govendor) anymore. To create a full workspace follow the guide below. To find out more about Go workspaces, visit the [official documentation](https://go.dev/ref/mod#workspaces).
```
Comment (dcs3spp) [inline tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
In Gateway version 5.1, the Gateway and plugins transitioned to a [Go modules](https://go.dev/ref/mod#introduction) build and don't use [Go vendor](https://pkg.go.dev/github.com/kardianos/govendor) anymore. To create a full workspace follow the guide below. To find out more about Go workspaces, visit the [official documentation](https://go.dev/ref/mod#workspaces).
```

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 2]
```diff
@@ -33,35 +33,35 @@ It's also possible to access the API definition data structure from within a plu
...
+To build compatible plugins, the gateway dependency that `tyk_plugin` uses needs to match the checkout in the tyk folder. To get the commit hash, go into `tyk` and run `git rev-parse HEAD`. To synchronize the gateway dependency go into `tyk_plugin` and run `go get github.com/TykTechnologies/tyk@<commit-hash>`.
+
+To gain a deeper understanding of `go work`, you can explore the [Go multi-module workspaces tutorial](https://go.dev/doc/tutorial/workspaces) on the official Go website.
 
-For more information about go workspaces, check out the [official
-tutorial](https://go.dev/doc/tutorial/workspaces).
 
-### Initialise plugin for gateway < 5.1
+### Initialise plugin for Gateway versions earlier than 5.1
+If you are using Tyk Gateway version that is older than 5.1, please use this section. The steps provided here are specific to Tyk Gateway versions below c5.1.
```
Comment (dcs3spp) [inline tyk-docs/content/plugins/supported-languages/golang.md:64]:
```suggestion
If you are using Tyk Gateway version that is older than 5.1, please use this section. The steps provided here are specific to Tyk Gateway versions below c5.1.
```

Comment (dcs3spp) [inline tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
If you are using a Tyk Gateway version that is older than 5.1, please use this section. The steps provided here are specific to Tyk Gateway versions below v5.1.
```

**Comment by letzya** [comment]
Thanks! That was damn fast !!!

**Pull request merged**